desc/plotting.py

```diff
@@ -3843,3 +3845,143 @@ def plot_logo(save_path=None, **kwargs):
         fig.savefig(save_path, facecolor=fig.get_facecolor(), edgecolor="none")
 
     return fig, ax
+
+
+def plot_adiabatic_invariant(
+    eq, rhos=None, alphas=None, num_pitch=None, pitch_idx=-1, mode="single-surface"
+):
+    """Plot the second adiabatic invariant J_|| (parallel action).
+
+    This function visualizes the parallel action invariant J_|| in either of two modes:
+    1. On a single flux surface as a function of fieldline label α and inverse
+       pitch angle 1/λ
+    2. As contours of J_|| for a fixed pitch angle across multiple flux surfaces
+
+    Parameters
+    ----------
+    eq : object
+        Equilibrium object containing magnetic field information
+    rhos : array_like or float, optional
+        Flux surface radii. If float, plots single surface.
+        Default: np.linspace(0.1, 1, 10)
+    alphas : array_like, optional
+        Fieldline label values (toroidal angle).
+        Default: np.linspace(0, 2π, 32, endpoint=True)
+    num_pitch : int, optional
+        Number of pitch angle values for bounce integral calculation.
+        Default: 16
+    pitch_idx : int, optional
+        Index of pitch angle to use for cross-section mode.
+        Default: -1 (last pitch angle)
+    mode : str, optional
+        Visualization mode, either "single-surface" or "cross-section".
+        Default: "single-surface"
+
+    Returns
+    -------
+    fig : matplotlib.figure.Figure
+        The figure object containing the plot
+    ax : matplotlib.axes.Axes
+        The axes object for further customization
+
+    Notes
+    -----
+    The second adiabatic invariant J_|| is calculated by integrating the parallel
+    velocity along the magnetic field line over a complete bounce period.
+    """
+    if rhos is None:
+        rhos = np.linspace(0.1, 1, 10)
+    elif isinstance(rhos, float):
+        print("Single rho provided...\n Plotting J_||(α, λ)")
+        rhos = np.array(
+            [rhos, 1.0]
+        )  # BUG: Array of len 1 takes all the mem and never finishes
+
+    if alphas is None:
+        alphas = np.linspace(0, 2 * np.pi, 32, endpoint=True)
+
+    if num_pitch is None:
+        num_pitch = 16
+
+    N_rho = len(rhos)
+    X, Y = 32, 64
```

Comment:
Add a comment for issue 1352 

`# TODO(#1352)`

to shield yourself from the demands that the other developers will surely request here

Reply:
Added.